### ISSUE TITLE
Clarification of treegrid aria-readonly

### ISSUE THREAD

**Issue by bjarketrux** [mdn/content#23496]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/treegrid_role

### What specific section or headline is this issue about?

#read-only_treegrids

### What information was incorrect, unhelpful, or incomplete?

The first two lines in the 'Read-only treegrids' section states:

> By default, treegrids are assumed to be editable. If a tree grid is not editable, use the [aria-readonly](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-readonly) attribute to inform assistive technologies the treegrid is readonly.

So assuming I only presents information in a table (which has a tree structure) then is seems that I should add `aria-readonly="true"` to the `treegrid` root element.

But the WAI-ARIA documentation for [treegrid](https://w3c.github.io/aria/#treegrid) states

> When the [aria-readonly](https://w3c.github.io/aria/#aria-readonly) attribute is applied to a focusable [gridcell](https://w3c.github.io/aria/#gridcell), it indicates whether the content contained in the [gridcell](https://w3c.github.io/aria/#gridcell) is editable.

Which makes it seems that it should only be used for editable (or disabled editable) content. Further it states:

> In a treegrid that provides content editing functions, if the content of a focusable [gridcell](https://w3c.github.io/aria/#gridcell) element is not editable, authors MAY set [aria-readonly](https://w3c.github.io/aria/#aria-readonly) to true on the [gridcell](https://w3c.github.io/aria/#gridcell) element. However, if a treegrid presents a collection of elements that do not support [aria-readonly](https://w3c.github.io/aria/#aria-readonly), such as a collection of [link](https://w3c.github.io/aria/#link) elements, it is not necessary for the author to specify a value for [aria-readonly](https://w3c.github.io/aria/#aria-readonly).

Which seems to state that it should not be necessary to specify on a information only `treegrid`.

[WAI-ARIA Authoring Practices 1.2](https://w3c.github.io/aria-practices/#treegrid) states more specifically:

> If the treegrid provides content editing functionality and contains cells that may have edit capabilities disabled in certain conditions, [aria-readonly](https://w3c.github.io/aria/#aria-readonly) is set to true on cells where editing is disabled. If edit functions are disabled for all cells, instead of setting aria-readonly to true on every cell, aria-readonly may be set to true on the treegrid element. **Treegrids that do not provide cell content editing functions do not include the aria-readonly attribute on any of their elements.**

This seems to say that for information content only `aria-readonly` should not be used.

Can someone please clarify if the MDN documentation is misleading or if my interpretation of the WAI_ARIA is wrong.
Thanks

### What did you expect to see?

Not sure but an alignment with the WAI-ARIA

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/accessibility/aria/roles/treegrid_role`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/treegrid_role
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/accessibility/aria/roles/treegrid_role/index.md
* Last commit: https://github.com/mdn/content/commit/22203bf02a971f0325a079245501350cc3341edf
* Document last modified: 2022-09-25T08:53:03.000Z

</details>

**Comment by Josh-Cena** [comment]
@estelle Since you wrote https://github.com/mdn/content/pull/13474, could you help here?

**Comment by estelle** [comment]
> So assuming I only presents information in a table (which has a tree structure) then is seems that I should add `aria-readonly="true"` to the `treegrid` root element.

ARIA should only be added if your using non-semantic elements to create semantic structure. The first rule of ARIA is if a native HTML element or attribute has the semantics and behavior you require, use it instead of re-purposing an element and adding ARIA. As you're using a table, which natively provides all the functionality required, neither the tree grid role nor the read-only attribute are needed nor should they be added. 

The entire ARIA section is for adding semantics when the HTML is not accessible. Create an accessible table and you don't need to add ARIA at all.

**Comment by Josh-Cena** [comment]
Let's suppose a non-semantic document, like a div-span soup. I have what's supposed to be a table and it's static and informational. I mark it as `treegrid`. Do I need `aria-readonly`? Or do I only need `aria-readonly` for editable components that happen to be disabled?

**Comment by estelle** [comment]
> Let's suppose a non-semantic document, like a div-span soup. I have what's supposed to be a table and it's static and informational. I mark it as `treegrid`. Do I need `aria-readonly`? Or do I only need `aria-readonly` for editable components that happen to be disabled?

If it's static and informational, the grid role should be used instead of treegrid. Treegrid isn't the right role.

**Comment by Josh-Cena** [comment]
But editable and collasible are two different things, right? The `grid` page says that if you have collapsible rows you should use `treegrid`, and the `treegrid` says it should be used for "editable or interactive" content. If I have a collapsible and non-editable grid, would that be `treegrid`? Would that be `aria-readonly`?

**Comment by bjarketrux** [comment]
Our purpose is like the email example from [ARIA Authoring Practices Guide (APG)](https://www.w3.org/WAI/ARIA/apg/): [Treegrid Email Inbox Example](https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/examples/treegrid-1/).

The example is interactive but not editable. They don't mention `aria-readonly`.

**Comment by Josh-Cena** [comment]
Yes @estelle. I did some research on this and it looks like `treegrid` has nothing to do with editing capabilities. As long as the grid has expand/collapse it would be a `treegrid`.
